iaso/models/org_unit.py

```diff
@@ -371,6 +371,13 @@ def as_dict_for_mobile(self):
             "aliases": self.aliases,
         }
 
+    def get_valid_parent_id(self, default_to_none: bool = True):
+        return (
+            self.parent_id
+            if self.parent_id is None or self.parent.validation_status == OrgUnit.VALIDATION_VALID
+            else (self.parent.get_valid_parent_id(default_to_none) if default_to_none is not True else None)
+        )
+
```

Comment:
@olethanh Even though I have added `.prefetch_related("parent")` to the `queryset`, I'm afraid this could result in a lot of queries to the database.
Could you advise if this is OK and, if not, what would be a better way?

Reply:
you can do a .prefetch_related("parent").prefech_related("parent_parent").. etc if you do 4 level it should be enough